client/cli/src/commands/metadata.rs

```diff
@@ -0,0 +1,55 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2021 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Key related CLI utilities
+
+use super::MetadataDump;
+use crate::{CliConfiguration, Error, SharedParams};
+use sc_client_api::UsageProvider;
+use sc_service::Arc;
+use sp_api::{Metadata, ProvideRuntimeApi};
+use sp_runtime::traits::Block as BlockT;
+use structopt::StructOpt;
+
+/// Cli subcommand for interacting with the runtime metadata.
+#[derive(Debug, StructOpt)]
+pub enum MetadataSubcommand {
+	/// Dump the runtime to a file or to stdout.
```

Comment:
Running `cargo run -- metadata dump --dev | less` I get the initial log lines intermingled with the actual metadata output. Not sure how common it is to use the stdout output. I'd be fine not mentioning it at all here and just support output to a file, but ideally our CLI commands should play well in a CLI environment.

Reply:
Log is printed to `stderr` and the actual output goes to `stdout`. So you can easily pipe.

Reply:
That's not what I'm seeing, the logs and the command output get mixed.

Reply:
Just do this: `cargo run -- metadata dump --dev > output`. I assume you can not decode the metadata hex with your bare eyes anyway, so less is not that useful :P